tests/test_core.py

```diff
@@ -341,67 +364,114 @@ def test_groupby_agg_dask(func, shape, array_chunks, group_chunks, add_nan, dtyp
     )
+    expected_groups = [0, 1, 2]
+    fill_value = False if func in ["all", "any"] else 123
 
-    expected, _ = groupby_reduce(array.compute(), labels, engine="numpy", **kwargs)
-    actual, _ = groupby_reduce(array.compute(), labels, engine=engine, **kwargs)
+    expected, _ = groupby_reduce(
+        array.compute(),
+        labels,
+        engine="numpy",
+        func=func,
+        expected_groups=expected_groups,
+        fill_value=fill_value,
+    )
+    actual, _ = groupby_reduce(
+        array.compute(),
+        labels,
+        engine=engine,
+        func=func,
+        expected_groups=expected_groups,
+        fill_value=fill_value,
+    )
     assert_equal(actual, expected)
 
     with raise_if_dask_computes():
-        actual, _ = groupby_reduce(array, labels, engine=engine, **kwargs)
+        actual, _ = groupby_reduce(
+            array,
+            labels,
+            engine=engine,
+            func=func,
+            expected_groups=expected_groups,
+            fill_value=fill_value,
+        )
     assert_equal(actual, expected)
 
     by = from_array(labels, group_chunks)
     with raise_if_dask_computes():
-        actual, _ = groupby_reduce(array, by, engine=engine, **kwargs)
+        actual, _ = groupby_reduce(
+            array,
+            by,
+            engine=engine,
+            func=func,
+            expected_groups=expected_groups,
+            fill_value=fill_value,
+        )
     assert_equal(expected, actual)
 
-    kwargs["expected_groups"] = [0, 2, 1]
+    expected_groups = [0, 2, 1]
     with raise_if_dask_computes():
-        actual, groups = groupby_reduce(array, by, engine=engine, **kwargs, sort=False)
+        actual, groups = groupby_reduce(
+            array,
+            by,
+            engine=engine,
+            func=func,
+            expected_groups=expected_groups,
+            fill_value=fill_value,
+            sort=False,
+        )
     assert_equal(groups, [0, 2, 1])
     assert_equal(expected, actual[..., [0, 2, 1]])
 
     kwargs["expected_groups"] = [0, 2, 1]
     with raise_if_dask_computes():
-        actual, groups = groupby_reduce(array, by, engine=engine, **kwargs, sort=True)
+        actual, groups = groupby_reduce(
+            array,
+            by,
+            engine=engine,
+            func=func,
+            expected_groups=expected_groups,
+            fill_value=fill_value,
+            sort=True,
+        )
     assert_equal(groups, [0, 1, 2])
     assert_equal(expected, actual)
 
 
-def test_numpy_reduce_axis_subset(engine):
+def test_numpy_reduce_axis_subset(engine: T_Engine) -> None:
     # TODO: add NaNs
     by = labels2d
     array = np.ones_like(by)
-    kwargs = dict(func="count", engine=engine, fill_value=0)
-    result, _ = groupby_reduce(array, by, **kwargs, axis=1)
+    result, _ = groupby_reduce(array, by, func="count", engine=engine, fill_value=0, axis=1)
```

Comment:
Lets revert this please. It makes it hard to see what the difference between the tests is

Reply:
dang. Lets just not type these tests then. It seems like a mypy bug?

Reply:
I guess using TypedDict would've worked?
 `kwargs: TypedDict_kwargs = dict(func="count", engine=engine, fill_value=0)` but they're not worth the effort.

To show which settings are important we could do something like:
```python
result, _ = groupby_reduce(
    array, 
    by, 
    axis=1, 
    engine=engine, 
    # Default settings:
    func="count", 
    fill_value=0,
)
```

Reply:
done